### PR TITLE
clarify docker plugin set docs

### PR DESCRIPTION
**Pull request by vieux**
related to https://github.com/docker/docker/issues/30795

ping @tiborvass @anusha-ragunathan

### CODE REVIEW

File: docs/reference/commandline/plugin_set.md [hunk 1]
```diff
@@ -34,6 +34,48 @@ The settings currently supported are:
...
+                }
+            ],
+	    "Mounts": [
+                {
+                    "Destination": "/mnt/state",
+                    "Name": "state",
+                    "Options": ["rbind"],
+                    "Settable": ["source"],
+                    "Source": "/var/lib/docker/plugins/",
+                    "Type": "bind"
```
Comment (anusha-ragunathan) [inline docs/reference/commandline/plugin_set.md]: Can you also add example sections for `Args` and `Devices` and explain them for completeness? w.r.t `Args`, please use a Name that's not `Args` :)

Comment (vieux) [inline docs/reference/commandline/plugin_set.md]: already added at the bottom, not enough ?

File: docs/reference/commandline/plugin_set.md [hunk 2]
```diff
@@ -91,12 +132,26 @@ $ docker plugin inspect -f '{{with $device := index .Settings.Devices 0}}{{$devi
...
 The following example change the source of the args on the `myplugin` plugin.
+Here is an extract of the plugin manifest:
+
+```
+{
+        "Config": {
+	    ...
+            "Args": {
+                "Name": "myargs",
+                "Settable": ["myargs"],
```
Comment (anusha-ragunathan) [inline docs/reference/commandline/plugin_set.md]: Should be `["Value"]`

Comment (vieux) [inline docs/reference/commandline/plugin_set.md]: yes, fixing

File: docs/reference/commandline/plugin_set.md (hunk 1 above)
Comment (anusha-ragunathan) [inline docs/reference/commandline/plugin_set.md]: Missing `Devices` section. Also it is worth clarifying that the `Settable` in the config is only the default.

**Comment by anusha-ragunathan** [comment]
Its still missing the `Devices` example.

**Comment by vieux** [comment]
@anusha-ragunathan PTAL

File: docs/reference/commandline/plugin_set.md [hunk 3]
```diff
@@ -34,6 +34,59 @@ The settings currently supported are:
...
+                    "name": "DEBUG",
+                    "settable": ["value"],
+                    "value": "0"
+                },
+                {
+                    "name": "LOGGING",
+                    "value": "1"
+                }
+            ],
+	    "devices": [
```
Comment (anusha-ragunathan) [inline docs/reference/commandline/plugin_set.md]: json needs some fixing. `devices` and `mounts` should be:

``` "devices": [
                {
                    "name": "mydevice",
                    "path": "/dev/foo",
                    "settable": ["path"]
                }
	   ],
"mounts": [
                {
                    "destination": "/baz",
                    "name": "mymount",
                    "options": ["rbind"],
                    "settable": ["source"],
                    "source": "/foo",
                    "type": "bind"
                }
            ]
```

File: docs/reference/commandline/plugin_set.md [hunk 4]
```diff
@@ -34,6 +34,59 @@ The settings currently supported are:
...
+            ],
+	    ...
+	 }
+}
+```
+
+In this example, we can see that the `value` of the `DEBUG` environment variable is settable,
+the `source` of the `mymount` mount is also settable. Same for the `path` of `mydevice` and `value` of `myargs`.
+
+On the contrary, the `LOGGING` environment variable doesn't have any settable field.
```
Comment (anusha-ragunathan) [inline docs/reference/commandline/plugin_set.md]: Good to add, `which implies that user cannot tweak it`

Comment (vieux) [inline docs/reference/commandline/plugin_set.md]: ok

**Comment by vieux** [comment]
@anusha-ragunathan should be good now

**Comment by anusha-ragunathan** [comment]
LGTM

**Pull request merged**